src/handlers/psp.ts

```diff
@@ -0,0 +1,436 @@
+        baseLogOptions
+      );
+      return;
+    }
+
+    enabledSubscriptions.push({
+      ...subscription,
+      id: subscriptionId,
+    });
+  });
+
+  const groupedSubscriptions: GroupedSubscriptions[] = [];
+  if (isEmpty(enabledSubscriptions)) {
+    node.logger.info('No proto-psp subscriptions to process', baseLogOptions);
+  } else {
+    const enabledSubscriptionsByTemplateId = groupBy(enabledSubscriptions, 'templateId');
+    Object.keys(enabledSubscriptionsByTemplateId).forEach((templateId) => {
+      // Get template details
+      const template = config.templates[templateId];
+      if (isNil(template)) {
+        node.logger.warn(`TemplateId ${templateId} not found in templates`, baseLogOptions);
+        return;
+      }
+      // Verify templateId
+      const expectedTemplateId = ethers.utils.solidityKeccak256(
+        ['bytes32', 'bytes'],
+        [template.endpointId, template.templateParameters]
+      );
+      if (expectedTemplateId !== templateId) {
+        node.logger.warn(`TemplateId ${templateId} does not match expected ${expectedTemplateId}`, baseLogOptions);
+        return;
+      }
+
+      // Get endpoint details
+      const endpoint = config.endpoints[template.endpointId];
+      if (isNil(endpoint)) {
+        node.logger.warn(`EndpointId ${template.endpointId} not found in endpoints`, baseLogOptions);
+        return;
+      }
+      // Verify endpointId
+      const expectedEndpointId = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(['string', 'string'], [endpoint.oisTitle, endpoint.endpointName])
+      );
+      if (expectedEndpointId !== template.endpointId) {
+        node.logger.warn(
+          `EndpointId ${template.endpointId} does not match expected ${expectedEndpointId}`,
+          baseLogOptions
+        );
+        return;
+      }
+
+      groupedSubscriptions.push({
+        subscriptions: enabledSubscriptionsByTemplateId[templateId],
+        template: { ...template, id: templateId },
+        endpoint: { ...endpoint, id: template.endpointId },
+      });
+    });
+  }
+
+  return {
+    config,
+    baseLogOptions,
+    groupedSubscriptions,
+    apiValuesBySubscriptionId: {},
+    providerStates: [],
+  };
+};
+
+const executeApiCalls = async (state: State): Promise<State> => {
+  const { config, baseLogOptions, groupedSubscriptions } = state;
+
+  // TODO: promise.all? 🤔
+  let apiValuesBySubscriptionId: { [subscriptionId: string]: ethers.BigNumber } = {};
+  for (const { subscriptions, template, endpoint } of groupedSubscriptions) {
+    const templateLogOptions: node.LogOptions = {
+      ...baseLogOptions,
+      additional: {
+        templateId: template.id,
+      },
+    };
+    const apiCallParameters = abi.decode(template.templateParameters);
+    const [errorCallApi, logsData] = await retryGo(() =>
+      callApi({
+        oises: config.ois,
+        apiCredentials: config.apiCredentials,
+        apiCallParameters,
```

Comment:
You do the API Call with `templateParameters`, but you also have parameters in Airkeeper's config (subscription.parameters) and it looks like those parameters are only used to get the `expectedSubscriptionId`? Are they the same parameters? Is this intentional?

Reply:
Subscriptions seems to allow parameters to be set [here](https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/protocol/StorageUtils.sol#L29) but they are being hardcoded to empty string [here](https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol#L274). Maybe @bbenligiray can better explain the use-case for `subscription.parameters`.